src/ppom-controller.ts

```diff
@@ -345,129 +361,216 @@ export class PPOMController extends BaseControllerV2<
-        continue;
-      }
-
-      // check if file is already in storage
-      if (
-        storageMetadata.find(
-          (file) =>
-            file.name === fileVersionInfo.name &&
-            file.chainId === fileVersionInfo.chainId &&
-            file.version === fileVersionInfo.version &&
-            file.checksum === fileVersionInfo.checksum,
-        )
-      ) {
-        continue;
-      }
-
-      const fileUrl = `${PPOM_CDN_BASE_URL}${fileVersionInfo.filePath}`;
-      const fileData = await this.#fetchBlob(fileUrl);
+    fileVersionInfo: PPOMFileVersion,
+  ) {
+    return storageMetadata.find(
+      (file) =>
+        file.name === fileVersionInfo.name &&
+        file.chainId === fileVersionInfo.chainId &&
+        file.version === fileVersionInfo.version &&
+        file.checksum === fileVersionInfo.checksum,
+    );
+  }
 
-      newFiles.push({
-        data: fileData,
-        ...fileVersionInfo,
-      });
+  /**
+   * Gets a single file from CDN and write to the storage.
+   *
+   * @param fileVersionInfo - Information about the file to be retrieved.
+   */
+  async #getFile(fileVersionInfo: PPOMFileVersion) {
+    const { storageMetadata } = this.state;
+    if (this.#checkFilePresentInStorage(storageMetadata, fileVersionInfo)) {
+      return;
     }
+    const fileUrl = `${PPOM_CDN_BASE_URL}${fileVersionInfo.filePath}`;
+    const fileData = await this.#fetchBlob(fileUrl);
 
-    return newFiles;
+    await this.#storage.writeFile({
+      data: fileData,
+      ...fileVersionInfo,
+    });
   }
 
-  /*
-   * Fetch the version info from the PPOM cdn.
-   *  update the version info in state.
+  /**
+   * As files for a chain are fetched this function set dataFetched property in chainIdCache to true.
+   *
+   * @param chainId - ChainId for which dataFetched is set to true.
    */
-  async #updateVersionInfo() {
-    const versionInfo = await this.#fetchVersionInfo(PPOM_VERSION_PATH);
-    if (versionInfo) {
+  #setChainIdDataFetched(chainId: string) {
+    const { chainIdCache } = this.state;
+    const chainIdObject = chainIdCache.find(
+      ({ chainId: cid }) => chainId === cid,
+    );
+    if (chainIdObject && !chainIdObject.dataFetched) {
       this.update((draftState) => {
-        draftState.versionInfo = versionInfo;
+        draftState.chainIdCache = [
+          { ...chainIdObject, dataFetched: true },
+          ...chainIdCache.filter(({ chainId: cid }) => chainId !== cid),
+        ];
       });
     }
   }
 
   /**
-   * Conditionally update the ppom configuration.
+   * Returns an array of new files that should be downloaded and saved to storage.
+   * The function is invoked if user if attempting transaction for a network,
+   * for which data is not previously fetched.
    *
-   * If the ppom configuration is out of date, this function will call `updatePPOM`
-   * to update the configuration.
+   * @returns A promise that resolves to an array of new files to download and save to storage.
```

Comment:
Is this comment incorrect?

It doesn't appear to return anything now and instead just downloads the files for the current chain.